### PR TITLE
Update mimir-prometheus

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR updates mimir-prometheus to latest main to get https://github.com/prometheus/prometheus/pull/11962 in.

I've checked changes in the libraries and nothing stands out, except some new options in [Alertmanager's ProxyConfig](https://github.com/grafana/mimir/blob/155424ae6b8d59222c622a9955d2b111c6fb359b/vendor/github.com/prometheus/common/config/http_config.go#L1088-L1103). Specifically new `proxy_from_environment` is now disabled for OAuth2 client (but not receiver config, see https://github.com/grafana/mimir/pull/2317 for rationale). Other settings in `ProxyConfig` are only active if `proxy_url` or `proxy_from_environment` is in use.

#### Checklist

- [na] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by jesusvazquez — APPROVED** [review]
I did look at the changes brought in from Prometheus and did not notice anything that could impact Mimir but I don't have all the context so I recommend someone else looking at it.

In the meantime I'm approving it because this brings my change about protecting the TSDB from an uninitialized wbl.

File: pkg/alertmanager/api.go [hunk 1]
```diff
@@ -440,12 +441,19 @@ func validateReceiverHTTPConfig(cfg commoncfg.HTTPClientConfig) error {
...
-		return errProxyURLNotAllowed
+	if cfg.OAuth2 != nil {
+		if cfg.OAuth2.ClientSecretFile != "" {
+			return errOAuth2SecretFileNotAllowed
+		}
+		// Mimir's "firewall" doesn't protect OAuth2 client, so we disallow Proxy settings here.
+		if cfg.OAuth2.ProxyURL.URL != nil {
+			return errProxyURLNotAllowed
+		}
+		if cfg.OAuth2.ProxyFromEnvironment {
```
Comment (pstibrany) [inline pkg/alertmanager/api.go:452]: No changelog for this as this is new in latest vendored Prometheus and we simply don't allow it.

File: pkg/alertmanager/api.go [hunk 2]
```diff
@@ -440,12 +441,19 @@ func validateReceiverHTTPConfig(cfg commoncfg.HTTPClientConfig) error {
...
-	if cfg.OAuth2 != nil && cfg.OAuth2.ClientSecretFile != "" {
-		return errOAuth2SecretFileNotAllowed
-	}
-	if cfg.OAuth2 != nil && cfg.OAuth2.ProxyURL.URL != nil {
-		return errProxyURLNotAllowed
+	if cfg.OAuth2 != nil {
+		if cfg.OAuth2.ClientSecretFile != "" {
+			return errOAuth2SecretFileNotAllowed
+		}
+		// Mimir's "firewall" doesn't protect OAuth2 client, so we disallow Proxy settings here.
```
Comment (colega) [inline pkg/alertmanager/api.go:448]: I don't know what "Mimir's firewall" is.

**Review by colega — APPROVED** [review]
LGTM, but should we get a review from someone from the alerting squad?

File: pkg/alertmanager/api.go (hunk 2 above)
Comment (pstibrany) [reply pkg/alertmanager/api.go:448]: It refers to this feature https://grafana.com/docs/mimir/latest/operators-guide/secure/securing-alertmanager/.

**Comment by pstibrany** [comment]
> should we get a review from someone from the alerting squad?

Good point! Thanks for review.

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

**Pull request merged**